dgv/harvester/DAG.py

```diff
@@ -7,135 +8,148 @@
+            pivot, value = "Lien_ancienne_admin", f"https://www.data.gouv.fr/fr/admin/harvester/{harvester['id']}"
+        else:
+            pivot, value = "harvester_id", harvester["id"]
+        rows = current_table.loc[current_table[pivot] == value]
+        # handling new harvesters
+        if len(rows) == 0:
+            new.append(harvester)
+            to_update = {
+                "harvester_id": harvester["id"],
+                "Lien_ancienne_admin": f"https://www.data.gouv.fr/fr/admin/harvester/{harvester['id']}",
+                "Organisation": harvester["owner_name"] or "",
+                "Lien_organisation": (
+                    f"https://www.data.gouv.fr/fr/{harvester['owner_type']}s/{harvester['owner_id']}/"
+                    if harvester["owner_type"] else ""
+                ),
+                "Statut": harvester["preview"],
+                "Statut_bizdev": ["L", "🆕 Nouveau"],
+            }
+            logging.info(f"New harvester: {harvester['id']}")
+        # handling existing ones
+        elif len(rows) == 1:
+            row = current_table.loc[current_table[pivot] == value].iloc[0]
+            if row["Statut"] != harvester["preview"]:
+                to_update = {"Statut": harvester["preview"]}
+            if pivot != "harvester_id":
+                # adding the id for future runs, it's cleaner
+                # and will be used to created other columns in grist
+                to_update["harvester_id"] = harvester["id"]
+            if to_update:
+                logging.info(f"Updating harvester: {harvester['id']}")
+        else:
+            issues.append(harvester)
+            logging.warning(f"Too many rows for harvester: {harvester['id']}")
+        update_records(
+            doc_id=doc_id,
+            table_id=table_id,
+            conditions={pivot: value},
+            new_values=to_update,
         )
+    ti.xcom_push(key="new", value=new)
+    ti.xcom_push(key="issues", value=issues)
 
-    text = text + "\nListe des moissonneurs en pending avec un autre statut : \n"
 
-    for lp in list_pendings_other:
-        text = (
-            text
-            + " - [{}]({}) - Moissonneur [{}]({}) - Lien vers [l'espace Admin]({}) \n".format(
-                lp["orga"], lp["orga_url"], lp["name"], lp["url"], lp["admin_url"]
-            )
-        )
+def publish_mattermost(ti):
+    pending_harvesters = ti.xcom_pull(key="harvesters_complete", task_ids="get_preview_state")
+    new = ti.xcom_pull(key="new", task_ids="fill_in_grist")
+    issues = ti.xcom_pull(key="issues", task_ids="fill_in_grist")
 
     text = (
-        f"{text}Avant validation, pensez à consulter [le pad des moissonneurs à laisser "
-        f"en attente de validation]({PAD_AWAITING_VALIDATION}) \n"
+        ":mega: Rapport hebdo sur l'état des moissonneurs en attente"
+        " ([lien grist](https://grist.numerique.gouv.fr/o/datagouv/6xrGmKARsDFR/Suivi-moissonneurs/p/3)):\n"
```

Comment:
I would link to [this view](https://grist.numerique.gouv.fr/o/datagouv/6xrGmKARsDFR/Suivi-moissonneurs/p/2) instead (only pending matters).